### PR TITLE
Add documentation for private deps

### PR DESCRIPTION
**Pull request by tucksaun**

### CODE REVIEW

File: cookbooks/private_deps.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+there should be no need to authenticate against a remote Git repository and
+there should be no authentication issue.
+
+But shall you need, this use case is also covered: SymfonyCloud generates one
+private SSH key per project and provide you with the public part allowing you to
+authorize it for Git repositories or specific machine accounts. This key is
+called a "deploy" key as its usage is reserved to deployment. GitHub provides
+`great documentation
+<https://developer.github.com/v3/guides/managing-deploy-keys/#machine-users>`_
+about this technic You can safely restrict its access to read-only operations.
```
[the remark this answers is not in the document]
Comment (tucksaun) [reply cookbooks/private_deps.rst]: Actually, it was meant to be two sentences.

File: cookbooks/private_dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
+Private Dependencies
+====================
+
+Some projects make use of private dependencies requiring authentication in order
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: Some projects use private ...

File: cookbooks/private_dependencies.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+for packages to be downloaded. While commiting those credentials in the Git
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: committing
these
in the source code Git repository

File: cookbooks/private_dependencies.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+repository source code is technically possible, not everyone is comfortable with
+this optional and this is generally considered as a bad practice.
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: technically possible, this is considered as a bad practice (remove everything in between)

File: cookbooks/private_dependencies.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+To handle this situation, two options are available:
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: You have two options:

File: cookbooks/private_dependencies.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+#. `Composer HTTP Authentication`_
+#. `SSH Deploy Key`_
+
+Composer HTTP Authentication
+----------------------------
+
+Composer allows you to define configuration for its `builtin authentication
+mechanism <https://getcomposer.org/doc/articles/http-basic-authentication.md>`_
+as a environment variable named ``COMPOSER_AUTH``. The content of the variable
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: as an

File: cookbooks/private_dependencies.rst [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    $ symfony var:set --sensitive --hide-at-runtime \
+    COMPOSER_AUTH='{"github-oauth": {"github.com": "oauthtoken"}, "http-basic": {"repo.example.org": {"username": "your-username", "password": "your-password"}}}'
+
+Once set, Composer will be able to authenticate and download dependencies from
+your authenticated repositories.
+
+SSH Deploy Key
+--------------
+
+Typically, private dependencies will be hosted in private Git repositories which
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: are hosted

File: cookbooks/private_dependencies.rst [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+requires SSH authentication. Fortunately, most private Composer tools (including
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: require

File: cookbooks/private_dependencies.rst [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+Satis, Toran Proxy, and `Private Packagist <https://packagist.com/>`_) mirror
+tagged releases of dependencies and serve them directly rather than hitting the
+Git repository. Therefore as long as your dependencies specify tagged releases
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: Therefore, as

Comment (fabpot) [reply cookbooks/private_dependencies.rst]: tagged releases, there

File: cookbooks/private_dependencies.rst [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+there should be no need to authenticate against a remote Git repository and
+there should be no authentication issue.
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: issues

File: cookbooks/private_dependencies.rst [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+But shall you need, this use case is also covered: SymfonyCloud generates one
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: you need it,

File: cookbooks/private_dependencies.rst [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+private SSH key per project and provide you with the public part allowing you to
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: provides you

Comment (fabpot) [reply cookbooks/private_dependencies.rst]: public key

File: cookbooks/private_dependencies.rst [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+authorize it for Git repositories or specific machine accounts. This key is
+called a "deploy" key as its usage is reserved to deployment. GitHub provides
+`documentation
+<https://developer.github.com/v3/guides/managing-deploy-keys/#machine-users>`_
+about this technic. You can safely restrict its access to read-only operations.
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: about this feature

**Comment by tucksaun** [comment]
@fabpot thank you for the review. all your comments should have been addressed.

File: cookbooks/private_dependencies.rst [hunk 14]
```diff
@@ -0,0 +1,58 @@
+Private Dependencies
+====================
+
+Some projects use private dependencies requiring authentication in order for
+packages to be downloaded. While committing these credentials in the source code
+Git repository source code is technically possible, this is considered as a bad
```
Comment (fabpot) [inline cookbooks/private_dependencies.rst]: double "source code", I would remove both.

Comment (tucksaun) [reply cookbooks/private_dependencies.rst]: good catch. And I totally agree. fixed

**Comment by jvanschie** [comment]
Thanks for adding this guys!

**Pull request merged**